dokka-subprojects/core/src/main/kotlin/org/jetbrains/dokka/transformers/documentation/DefaultDocumentableMerger.kt

```diff
@@ -106,9 +106,20 @@ public class DefaultDocumentableMerger(context: DokkaContext) : DocumentableMerg
                 merged as List<T>
             }
 
+        fun processClashingElements(elements: List<T>) =
+            elements.map { it to it.sourceSets }
+                .groupBy { it.first.dri }
+                .values
+                .flatMap(::mergeClashingElements)
 
         fun analyzeExpectActual(sameDriElements: List<T>): List<T> {
             val (expects, actuals) = sameDriElements.partition { it.expectPresentInSet != null }
+            // It's possible that there are no `expect` declarations, but there are `actual` declarations,
```

Comment:
LGTM
But I am not sure about treating such declarations as clashing. 
It works for stdlib since `mergeImplicitExpectActualDeclarations=true` is enabled. In other projects, the documentation would have separated pages that can look inconsistent regarding normal expect/actual declarations. 
Is there a reason to have clashing declarations instead of a merged one?

Reply:
Yeah, good idea. Will see what can be done here.